### ISSUE TITLE
False positive in Style/HashExcept for Array of Arrays, and Auto-correct breaks code

### ISSUE THREAD

**Issue by strviola** [rubocop/rubocop#14079]
When there is a process to exclude elements using the `reject` method with 2 block arguments for an Array whose elements are Arrays of length 2, `Style/HashExcept` reacts.

```ruby
arr = [[:foo, 1], [:bar, 2]]
arr2 = arr.reject { |e1, _e2| e1 == :bar }
p arr2
```

--------

## Expected behavior

Since there is no more concise implementation possible, Rubocop should not issue a warning.

## Actual behavior

`Style/HashExcept` is triggered. Additionally, when the `-A` option is added, it converts the code to the following, but since the except method does not exist for Array, a NoMethodError occurs.

```ruby
arr = [[:foo, 1], [:bar, 2]]
arr2 = arr.except(:bar) # ERROR
p arr2
```

## Steps to reproduce the problem

```
$ cat test.rb 
arr = [[:foo, 1], [:bar, 2]]
arr2 = arr.reject { |e1, _e2| e1 == :bar }
puts arr2

$ ruby test.rb 
[[:foo, 1]]

$ rubocop test.rb 
Inspecting 1 file
C

Offenses:

test.rb:2:12: C: [Correctable] Style/HashExcept: Use except(:bar) instead.
arr2 = arr.reject { |e1, _e2| e1 == :bar }
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable

$ rubocop -A test.rb 
Inspecting 1 file
C

Offenses:

test.rb:2:12: C: [Corrected] Style/HashExcept: Use except(:bar) instead.
arr2 = arr.reject { |e1, _e2| e1 == :bar }
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected

$ cat test.rb
arr = [[:foo, 1], [:bar, 2]]
arr2 = arr.except(:bar)
p arr2

$ ruby test.rb
test.rb:2:in `<main>': undefined method `except' for [[:foo, 1], [:bar, 2]]:Array (NoMethodError)

arr2 = arr.except(:bar)
          ^^^^^^^
```

## RuboCop version

```
ruby 3.2.7
rubocop (1.75.1)
rubocop-ast (1.43.0)
rubocop-rspec (3.5.0)
```

**Comment by koic** [comment]
`Style/HashExcept` cop is already marked as unsafe, and the reason for this is documented:
https://docs.rubocop.org/rubocop/1.75/cops_style.html#stylehashexcept
